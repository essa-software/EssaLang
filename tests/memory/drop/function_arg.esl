func foo(arg: string) {
    print("{}\n", arg);
    print("before drop\n");
}

func main() {
    foo(string_new("dtor0"));
    print("after drop\n");
}
