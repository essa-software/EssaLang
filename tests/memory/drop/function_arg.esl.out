dtor0
before drop
dropped string: dtor0
after drop
